src/jit/gentree.cpp

```diff
@@ -8296,7 +8296,7 @@ bool GenTree::gtSetFlags() const
 #endif
 
 #else // !LEGACY_BACKEND
-#ifdef _TARGET_XARCH_
+#if defined(_TARGET_XARCH_) || defined(_TARGET_ARM64_)
```

Comment:
The comments in this method need a serious overhaul, including a better explanation of why the code for `FEATURE_SET_FLAGS` and `!FEATURE_SET_FLAGS && ~LEGACY_BACKEND` is the same.
However, I suppose that's a fix for another PR.

Reply:
That method should be a giant `#if LEGACY_BACKEND`/`#else`/`#endif` where `#else` contains just `(((gtFlags & GTF_SET_FLAGS) != 0) && (gtOper != GT_IND))`. For some reason it keeps getting patched up and it every time it becomes more convoluted.